### PR TITLE
215 fix swimlane examples

### LINKED ISSUE #215: [Bug]: improper example for tm_g_swimlane
### What happened?

When running example, you see

image

Clearly `TRTDURD` does no longer exist in `ADSL`. If you change `TRTDURD` to the second common column `DTHADY` that is an integer that exist in both `ADSL` and `ADRS`, the plot never gets plotted. And there is no other common integer columns besides `EOSDY` for which this plot never gets plotted.

```{R}
library(dplyr)
library(nestcolor)

ADSL <- rADSL
ADRS <- rADRS

ADRS <- ADRS %>%
  dplyr::filter(PARAMCD == "LSTASDI" & DCSREAS == "Death") %>%
  mutate(AVALC = DCSREAS, ADY = EOSDY) %>%
  base::rbind(ADRS %>% dplyr::filter(PARAMCD == "OVRINV" & AVALC != "NE")) %>%
  arrange(USUBJID)

app <- init(
  data = cdisc_data(
    cdisc_dataset("ADSL", ADSL, code = "ADSL <- rADSL"),
    cdisc_dataset("ADRS", ADRS,
      code = "ADRS <- rADRS
              ADRS <- ADRS %>% dplyr::filter(PARAMCD == 'LSTASDI' & DCSREAS == 'Death') %>%
                      mutate(AVALC = DCSREAS, ADY = EOSDY) %>%
              rbind(ADRS %>% dplyr::filter(PARAMCD == 'OVRINV' & AVALC != 'NE')) %>%
              arrange(USUBJID)"
    ),
    check = TRUE
  ),
  modules = modules(
    tm_g_swimlane(
      label = "Swimlane Plot",
      dataname = "ADRS",
      bar_var = teal.transform::choices_selected(
        selected = "TRTDURD",
        choices = c("TRTDURD", "EOSDY")
      ),
      bar_color_var = teal.transform::choices_selected(
        selected = "EOSSTT",
        choices = c("EOSSTT", "ARM", "ARMCD", "ACTARM", "ACTARMCD", "AGEGR1", "SEX")
      ),
      sort_var = teal.transform::choices_selected(
        selected = "ACTARMCD",
        choices = c("USUBJID", "SITEID", "ACTARMCD", "TRTDURD")
      ),
      marker_pos_var = teal.transform::choices_selected(
        selected = "ADY",
        choices = c("ADY")
      ),
      marker_shape_var = teal.transform::choices_selected(
        selected = "AVALC",
        c("AVALC", "AVISIT")
      ),
      marker_shape_opt = c("CR" = 16, "PR" = 17, "SD" = 18, "PD" = 15, "Death" = 8),
      marker_color_var = teal.transform::choices_selected(
        selected = "AVALC",
        choices = c("AVALC", "AVISIT")
      ),
      marker_color_opt = c(
        "CR" = "green", "PR" = "blue", "SD" = "goldenrod",
        "PD" = "red", "Death" = "black"
      ),
      vref_line = c(30, 60),
      anno_txt_var = teal.transform::choices_selected(
        selected = c("ACTARM", "SEX"),
        choices = c(
          "ARM", "ARMCD", "ACTARM", "ACTARMCD", "AGEGR1",
          "SEX", "RACE", "COUNTRY", "DCSREAS", "DCSREASP"
        )
      )
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

### sessionInfo()

```R
R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Europe.utf8  LC_CTYPE=English_Europe.utf8   
[3] LC_MONETARY=English_Europe.utf8 LC_NUMERIC=C                   
[5] LC_TIME=English_Europe.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nestcolor_0.1.2.9001      teal.osprey_0.1.15.9014   teal_0.13.0.9006         
 [4] teal.transform_0.3.0.9003 magrittr_2.0.3            teal.data_0.2.0.9003     
 [7] shiny_1.7.4.1             osprey_0.1.15.9011        ggplot2_3.4.2            
[10] dplyr_1.1.2              

loaded via a namespace (and not attached):
 [1] sass_0.4.6               tidyr_1.3.0              jsonlite_1.8.7          
 [4] splines_4.2.3            R.utils_2.12.2           bslib_0.5.0             
 [7] teal.slice_0.3.0.9005    Rdpack_2.4               teal.code_0.3.0.9006    
[10] yaml_2.3.7               formatters_0.5.0.9002    pillar_1.9.0            
[13] backports_1.4.1          lattice_0.20-45          rtables_0.6.1.9012      
[16] glue_1.6.2               teal.widgets_0.3.0.9003  digest_0.6.31           
[19] promises_1.2.0.1         rbibutils_2.2.13         checkmate_2.2.0         
[22] colorspace_2.1-0         htmltools_0.5.5          httpuv_1.6.11           
[25] Matrix_1.5-3             R.oo_1.25.0              pkgconfig_2.0.3         
[28] broom_1.0.5              logger_0.2.2             purrr_1.0.1             
[31] xtable_1.8-4             scales_1.2.1             fontawesome_0.5.1       
[34] later_1.3.1              tibble_3.2.1             styler_1.10.1           
[37] farver_2.1.1             generics_0.1.3           ellipsis_0.3.2          
[40] cachem_1.0.8             withr_2.5.0              shinyjs_2.1.0           
[43] shinyvalidate_0.1.2      cli_3.6.1                crayon_1.5.2            
[46] survival_3.5-3           mime_0.12                memoise_2.0.1           
[49] evaluate_0.21            R.methodsS3_1.8.2        fansi_1.0.4             
[52] R.cache_0.16.0           textshaping_0.3.6        teal.reporter_0.1.1.9016
[55] rsconnect_0.8.29         tools_4.2.3              lifecycle_1.0.3         
[58] munsell_0.5.0            compiler_4.2.3           jquerylib_0.1.4         
[61] systemfonts_1.0.4        rlang_1.1.1              grid_4.2.3              
[64] rstudioapi_0.15.0        shinyWidgets_0.7.6       rmarkdown_2.23          
[67] gtable_0.3.3             teal.logger_0.1.1.9007   R6_2.5.1                
[70] knitr_1.43               fastmap_1.1.1            utf8_1.2.3              
[73] ragg_1.2.5               tern_0.8.4.9004          Rcpp_1.0.11             
[76] vctrs_0.6.2              tidyselect_1.2.0         xfun_0.39
```

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by m7pr**
Closes #215

### CODE REVIEW

**Comment by m7pr** [comment]
@lcd2yyz would you do an honor of reviewing and accepting the PR?

**Comment by m7pr** [comment]
Hey @cicdguy would you be able to help with the GHA build in here? It fails becuase the package can not be installed, and the further logs are written to some file that I can not track. https://github.com/insightsengineering/teal.osprey/actions/runs/5599593492/jobs/10240766966#step:37:75 Would you be able to point to the file? Thank!

> * checking whether package ‘teal.osprey’ can be installed ... ERROR
Installation failed.
See ‘/__w/teal.osprey/teal.osprey/teal.osprey.Rcheck/00install.out’ for details.

File: NAMESPACE [hunk 1]
```diff
@@ -19,4 +19,6 @@ import(formatters)
 import(osprey)
 import(shiny)
 import(teal)
+importFrom(teal.reporter,simple_reporter_srv)
+importFrom(teal.reporter,simple_reporter_ui)
```
Comment (gogonzo) [inline NAMESPACE:23]: all `simple_reporter_srv` and `simple_reporter_ui` are prefixed by `teal.reporter::`. Why do we need this import?

Comment (m7pr) [reply NAMESPACE:23]: I was trying to fix this WARNING in R CMD CHECK (for different function back then) https://github.com/insightsengineering/teal.osprey/actions/runs/5598928237/jobs/10239237480#step:37:96 
Will try to update without imports, to triggers this again

Comment (m7pr) [reply NAMESPACE]: reverted in 0017981490637da404771895fedf70a540246c0e - let's see the GHA build status

Comment (m7pr) [reply NAMESPACE]: oh my, all check have passed

**Review by gogonzo — APPROVED** [review]

**Pull request merged**